index/src/executors/bridge_executors/phala_xtransfer.rs

```diff
@@ -0,0 +1,206 @@
+use pink_subrpc::{create_transaction, send_transaction, ExtraParam};
+
+use crate::traits::{common::Error, executor::BridgeExecutor};
+use crate::utils::ToArray;
+use crate::AccountType;
+use alloc::{
+    string::{String, ToString},
+    vec::Vec,
+};
+use scale::Decode;
+
+use xcm::v1::{prelude::*, AssetId, Fungibility, Junctions, MultiAsset, MultiLocation};
+
+pub type Phala2AcalaExecutor = PhalaXTransferExecutor;
+pub type Phala2MoonbeamExecutor = PhalaXTransferExecutor;
+
+#[derive(Clone)]
+pub struct PhalaXTransferExecutor {
+    rpc: String,
+    dest_chain_id: u32,
+    // account32 or account20
+    account_type: AccountType,
+}
+
+#[allow(dead_code)]
+impl PhalaXTransferExecutor {
+    pub fn new(rpc: &str, dest_chain_id: u32, account_type: AccountType) -> Self
+    where
+        Self: Sized,
+    {
+        Self {
+            rpc: rpc.to_string(),
+            dest_chain_id,
+            account_type,
+        }
+    }
+}
+
+impl BridgeExecutor for PhalaXTransferExecutor {
+    fn transfer(
+        &self,
+        signer: [u8; 32],
+        asset: Vec<u8>,
+        recipient: Vec<u8>,
+        amount: u128,
+        extra: ExtraParam,
+    ) -> core::result::Result<Vec<u8>, Error> {
+        let asset_location: MultiLocation =
+            Decode::decode(&mut asset.as_slice()).map_err(|_| Error::InvalidMultilocation)?;
+        let multi_asset = MultiAsset {
+            id: AssetId::Concrete(asset_location),
+            fun: Fungibility::Fungible(amount),
+        };
+        let dest = MultiLocation::new(
+            1,
```

Comment:
As I mentioned above, if we transfer asset to Ethereum, the `dest` should be a `(0, X3(...))`.